topk-js/src/control/field_spec.rs

```diff
@@ -0,0 +1,140 @@
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+
+use super::{
+    data_type::DataType,
+    field_index::{FieldIndex, VectorDistanceMetric},
+};
+
+#[napi]
+#[derive(Clone)]
+pub struct FieldSpec {
+    data_type: DataType,
+    required: bool,
+    index: Option<FieldIndex>,
+}
+
+#[napi]
+impl FieldSpec {
+    pub fn new(data_type: DataType) -> Self {
+        Self {
+            data_type,
+            required: false,
+            index: None,
+        }
+    }
+
+    #[napi]
+    pub fn required(&self) -> Self {
+        Self {
+            required: true,
+            ..self.clone()
+        }
+    }
+
+    #[napi]
+    pub fn optional(&self) -> Self {
+        Self {
+            required: false,
+            ..self.clone()
+        }
+    }
+
+    #[napi]
+    pub fn keyword_index(&self) -> Self {
```

Comment:
these shouldn't exist, the syntax is `f32Vector().index(vectorIndex({metric: "..."}))`